posts/2022-01-14-construct-vendia-univseral-application-using-domain-driven-design.md

```diff
@@ -0,0 +1,177 @@
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/96793170/151029167-ac2d28b7-7861-4790-a972-5a7a0ea50156.png" />
+</p>
+<p align="center">Figure 2 - Bounded Context Reference Architecture</p>
+
+Listed below are best practices for building services and data models for a bounded context.
+
+
+* The aggregate is composed of multiple Entities and Value Objects(singletons without unique identifiers). Try not to pack more than 3 Aggregates in a service boundary. 
+* An aggregate defines a transaction consistency boundary around one or more entities.
+* An aggregate has a top-level entity called a root entity. Lookup on aggregate records is done using the root entity identifier. Any other entities in the aggregate are children of the root and are referenced using pointers. 
+* A service should not be larger than a bounded context, and it should not be smaller than an aggregate. 
+* For cross-domain data sync with external systems, use an async eventing system in your organization.
+
+Hopefully, this gives you a good introduction or refresher on DDD. Next, we develop an understanding of Uni structure, terms, and concepts. 
+
+
+## Uni Development Lifecycle
+
+Uni is a cross-cloud, cross-party, cross-account, cross-region data and code sharing application powered by Vendia's serverless distributed ledger technology. A uni is composed of [nodes](https://www.vendia.net/docs/share/terms-and-definitions#node) modeling [participants](https://www.vendia.net/docs/share/terms-and-definitions#participants), each of which possesses an ordered, replicated data model in the form of a distributed [ledger](https://www.vendia.net/docs/share/terms-and-definitions#ledger).
+
+A Uni software development life cycle has five stages: Model, Design, Deploy, Operate, and Scale. Since a Uni is serverless, deployment, operations, and scaling are easy. Once you have the schema design, deployment is just a couple of clicks, and operations are handled by Vendia.  Vendia is responsible for scalability and performance. This leaves us with data modeling and design. Data models are the core of any software. In Uni, the design phase involves constructing a schema that is a deployable artifact and leads to the creation of Uni. In this blog, we will focus on data modeling and design.
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/96793170/151029339-3c97b4d4-e66a-4b37-ae21-6200e22cb042.png" />
+</p>
+<p align="center">Figure 3 - Uni Development Lifecycle Stages</p>
+
+
+## Uni Schema Anatomy
+
+Let’s imagine a simple Uni schema that provides a datastore for core customer data attributes. Figure 4 breaks the Uni schema into parts and tries to map it to concepts you use in a relational database. This will help you quickly get started on building your own Uni. 
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/96793170/151029529-fa688c30-766c-4803-bc59-3b953e3ae0c4.png" />
```

Comment:
Not sure how closely you want the names in the figure to match reality. Keep in mind the queries and mutations generated by the compiler. For example, `Customers` will generate queries like `list_CustomersItems`. Using a singular `Customer` will generate more natural queries like `list_CustomerItems`.